single_kernel_mongo/utils/mongo_connection.py

```diff
@@ -626,3 +662,8 @@ def is_shard_aware(self, shard_name: str) -> bool:
                 return shard["state"] == SHARD_AWARE_STATE
 
         return False
+
+    def are_all_shards_aware(self) -> bool:
+        """Returns True if all shards are shard aware."""
+        sc_status = self.client.admin.command("listShards")
+        return all(shard["state"] == SHARD_AWARE_STATE for shard in sc_status["shards"])
```

Comment:
Note: this checks that all shards in `sc_status` are shard aware, in theory it could be possible that we have a shard that is integrated to the config-server but is not yet in `sc_status`

Reply:
This is correct.
I'll rework this in a better way :)

Reply:
ack! Thank you!